sources/web/datalab/polymer/modules/file-manager/file-manager.ts

```diff
@@ -206,16 +208,73 @@ interface FileManager {
    * Returns the url to open the given file in the notebook editor.
    * @param fileId id for the file to open in the notebook editor.
    */
-  getNotebookUrl(file: DatalabFileId): Promise<string>;
+  getNotebookUrl(fileId: DatalabFileId): Promise<string>;
 
   /**
    * Returns the url to open the given file in the text editor.
    * @param fileId id for the file to open in the text editor.
    */
-  getEditorUrl(file: DatalabFileId): Promise<string>;
+  getEditorUrl(fileId: DatalabFileId): Promise<string>;
 
   /**
    * Creates a path history from a path string.
    */
   pathToPathHistory(path: string): DatalabFile[];
 }
+
+/**
+ * Base implementation of the FileManager interface that contains common
+ * functionality for the different FileManager classes.
+ */
+class BaseFileManager {
```

Comment:
Doesn't this need to have "implements FileManager"?

Reply:
It does. Fixed.